### PR TITLE
Migration of listent function for fastify v4

### LINKED ISSUE #1401: [Bug]: Migration of the variadic signature for fastify
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Current Behavior

When we did the migration to Fastify 4, we didn't change the `listent` variadic signature as noted in the [migration guide](https://www.fastify.io/docs/latest/Guides/Migration-Guide-V4/#deprecation-of-variadic-listen-signature) so we get a warn error every time the backend is up.

### Expected Behavior

No warning method displayed in the backend.

### Steps To Reproduce

1. Start the backend

### Workaround (if any)

_No response_

### What browsers are you seeing the problem on?

_No response_

### Open Data Hub Version

_No response_

### Anything else

_No response_

### PR DESCRIPTION
**Pull request by lucferbux**
## Description

Closes #1401 

## How Has This Been Tested?

1. Runing the backend
2. Check that no longer displays the warn about variadic call signature

## Request review criteria:

- [X] The commits have meaningful messages (squashes happen on merge by the bot).
- [ ] Included any necessary screenshots or gifs if it was a UI change.
- [ ] Included tags to the UX team if it was a UI/UX change.
- [X] Testing instructions have been added in the PR body (for PRs involving changes that are not immediately obvious).
- [X] The developer has manually tested the changes and verified that the changes work
- [ ] The developer has added tests or explained why testing cannot be added (unit tests & storybook for related changes)

### CODE REVIEW

**Comment by manaswinidas** [comment]
/lgtm - The changes work for me

Before
Screenshot 2023-06-16 at 4 05 44 PM

After
Screenshot 2023-06-16 at 4 08 44 PM

**Review by manaswinidas — APPROVED** [review]

File: backend/src/server.ts [hunk 1]
```diff
@@ -19,7 +19,7 @@ const app = fastify({
 
 app.register(initializeApp);
 
-app.listen(PORT, IP, (err) => {
+app.listen({ port: Number(PORT), host: IP}, (err) => {
```
Comment (andrewballantyne) [inline backend/src/server.ts:22]: Weird to translate to Number here... It's an env var, we should purify it at source.

**Review by andrewballantyne — CHANGES_REQUESTED** [review]

File: backend/src/server.ts (hunk 1 above)
Comment (DaoDaoNoCode) [reply backend/src/server.ts:22]: I agree, try to process the `PORT` here https://github.com/opendatahub-io/odh-dashboard/blob/main/backend/src/utils/constants.ts#L5

Comment (lucferbux) [reply backend/src/server.ts:22]: Ok, yup, let me do that in the env var!

**Review by DaoDaoNoCode — APPROVED** [review]
lgtm

**Review by Gkrumbach07 — APPROVED** [review]
re-reviewed. and approved

LGTM

File: backend/src/server.ts (hunk 1 above)
Comment (Gkrumbach07) [reply backend/src/server.ts]: resolved

**Comment by Gkrumbach07** [comment]
I dont know if the bot will merge unless Andrew personally resolves his comment @andrewballantyne

**Pull request merged**